renderapi/transform/leaf/thin_plate_spline.py

```diff
@@ -3,9 +3,24 @@
 from renderapi.utils import encodeBase64, decodeBase64
 from .transform import Transform
 import scipy.spatial
+import logging
+import sys
 __all__ = ['ThinPlateSplineTransform']
 
 
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler(sys.stdout))
+
+
+class AdaptiveMeshEstimationError(Exception):
```

Comment:
sorry to be nitpicky, but this should subclass EstimationError, or we can make EstimationError have an keyword argument of transform=None, so that it works whether you want to pass on the transform calculated or not, but code can be written to generically catch all EstimationErrors

Reply:
that's fine, I can't quite imagine a use case where you would/should use a failed result... but you seem to. So, I don't mind making it what you're after.

Reply:
oh! well.. actually I was intuiting because you had it as a warning before that you had a use case in mind to use the transform, so i was trying to accommodate that.

Reply:
In a scipy iterative solver situation, I've come to expect the behavior that a solution is always returned, but a max iter message pops up. That's what I was originally going for with the logger message:
```
from scipy.optimize import minimize, rosen, rosen_der
x0 = [1.3, 0.7, 0.8, 1.9, 1.2]
res = minimize(rosen, x0, method='Nelder-Mead', tol=1e-6, options={'maxiter': 1000, 'disp': True})
Optimization terminated successfully.
         Current function value: 0.000000
         Iterations: 295
         Function evaluations: 494

print(res.__class__)
<class 'scipy.optimize.optimize.OptimizeResult'>

del res

res = minimize(rosen, x0, method='Nelder-Mead', tol=1e-6, options={'maxiter': 10, 'disp': True})
Warning: Maximum number of iterations has been exceeded.

print(res.__class__)
<class 'scipy.optimize.optimize.OptimizeResult'>
```

Reply:
but maybe a better precedent is just bombing out like in our polynomial transform:
https://github.com/fcollman/render-python/blob/d5ccfafd75e5a3694e989e9c97397d315c9b0c52/renderapi/transform/leaf/polynomial_models.py#L231-L234

Reply:
yeah i like bombing out better personally.